homeassistant/components/izone/climate.py

```diff
@@ -0,0 +1,468 @@
+"""
+Support for the iZone HVAC.
+
+For more details about this platform, please refer to the documentation at
+https://github.com/Swamp-Ig/izone_custom_component
+"""
+import logging
+from typing import cast
+
+from pizone import Zone, Controller
+
+from ..climate import ClimateDevice
+from ..climate.const import (
+    STATE_AUTO, STATE_COOL, STATE_DRY, STATE_FAN_ONLY, STATE_HEAT,
+    STATE_ECO, SUPPORT_FAN_MODE, SUPPORT_ON_OFF,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from ...const import (
+    ATTR_TEMPERATURE, PRECISION_HALVES, TEMP_CELSIUS,
+    STATE_OFF, STATE_ON)
+from ...helpers.temperature import display_temp as show_temp
+from ...helpers.typing import ConfigType, HomeAssistantType
+
+from .constants import DATA_ADD_ENTRIES, DATA_DISCOVERY_SERVICE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, config: ConfigType,
+                            async_add_entities):
+    """Initialize an IZone Controller."""
+    disco = hass.data[DATA_DISCOVERY_SERVICE]
+
+    # create any components not yet created
+    for controller in disco.controllers.values():
+        disco.init_controller(controller, async_add_entities)
+
+    # disco will use the register function to register any further components
+    hass.data[DATA_ADD_ENTRIES] = async_add_entities
+
+    return True
+
+
+class ControllerDevice(ClimateDevice):
+    """Representation of iZone Controller."""
+
+    def __init__(self, controller, async_add_entities) -> None:
+        """Initialise ControllerDevice."""
+        self._controller = cast(Controller, controller)
+
+        self._supported_features = (SUPPORT_OPERATION_MODE |
+                                    SUPPORT_FAN_MODE | SUPPORT_ON_OFF)
+
+        if ((controller.ras_mode == 'master' and controller.zone_ctrl == 13) or
+                controller.ras_mode == 'RAS'):
+            self._supported_features |= SUPPORT_TARGET_TEMPERATURE
+
+        self._state_to_pizone = {
+            STATE_COOL: Controller.Mode.COOL,
+            STATE_HEAT: Controller.Mode.HEAT,
+            STATE_AUTO: Controller.Mode.AUTO,
+            STATE_FAN_ONLY: Controller.Mode.VENT,
+            STATE_DRY: Controller.Mode.DRY,
+        }
+        if controller.free_air_enabled:
+            self._state_to_pizone[STATE_ECO] = Controller.Mode.FREE_AIR
+
+        self._fan_to_pizone = {}
+        for fan in controller.fan_modes:
+            self._fan_to_pizone[fan.name.title()] = fan
+
+        self._device_info = {
+            'identifiers': {
+                (DOMAIN, self.unique_id)
+            },
+            'name': self.name,
+            'manufacturer': 'IZone',
+            'model': self._controller.sys_type,
+        }
+        async_add_entities([self], True)
```

Comment:
Why does this use `, True` if the device is non-polling?

Reply:
Fixed.